src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -351,6 +351,7 @@ private ZonedDateTime createZonedDateTime(int timeInSeconds) {
    */
   private boolean includeTransferInItinerary(Leg transitLegBeforeTransfer) {
     return (
+      transitLegBeforeTransfer == null ||
```

Comment:
Question: Is this just hiding the problem? A transfer may exist, it is just that there is not a regular transit leg in front of it - it is a FLEX transit arriving onBoard. To fix this we must allow a transfer between access and transit. The same thing apply to transit + egress. I think several places in the code assume the same thing. 

To test this we need to add test-cases in the modules-test for Raptor, optimized-transfer and itinerary mapper.

Reply:
@vpaturet We talked about this in the meeting today and came to this conclusion: we don't need to make the case of Flex -> Transit work so leaving a comment about making this work if we want those cases taking into consideration is enough.

Reply:
Added comment